File: tests/testWCS2.py
from astropy.io import fits
from astropy.wcs import WCS
import numpy as np

# get a user home directory
from pathlib import Path

home = str(Path.home())

header = fits.open(home + "/Downloads/SUPM19EAEFCA02F0C500.fits")[0].header
print(header)

wcs = WCS(header)
print(wcs)

pixcrd = np.array(
    [[937.6906124764719, 7012.886595879788], [2235.7681311858246, 6188.131977699766]],
    dtype=np.float64,
)

print(pixcrd)

world = wcs.wcs_pix2world(pixcrd, 0)
print(world)

File: tests/testWCS2.jl
using FITSIO
using WCS

# Read in the FITS file
file = homedir() * "/Downloads/SUPM19EAEFCA02F0C500.fits"
f = FITS(file)

# read-in the FITS header
header = read_header(f[1], String)

# create a WCS object
wcs_array = WCS.from_header(header)
wcs = wcs_array[1]

# print the WCS object
println(wcs)

pix = [937.6906124764719, 7012.886595879788]
pix2 = [2235.7681311858246, 6188.131977699766]